pkg/sql/delegate/show_types.go

```diff
@@ -0,0 +1,25 @@
+// Copyright 2020 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package delegate
+
+import "github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+
+func (d *delegator) delegateShowTypes() (tree.Statement, error) {
+	// TODO (SQL Features, SQL Exec): Once more user defined types are added
+	//  they should be added here.
+	return parse(`
+SELECT
+  schema, name, value AS description
+FROM
+  [SHOW ENUMS]
```

Comment:
IMHO I'd have done it the other way around: make SHOW TYPES add a column with the kind of the type (enum, domain etc)

then make SHOW ENUMS filter with `WHERE kind = 'enum'`

Reply:
isn't there an `information_schema` that would report on that anyway?

Reply:
I'm not sure -- the show enums query is powered by the pg_catalog tables. My thought on the matter is that this way is more straightforward. The ways that different kinds are collected are pretty different. It would be simpler to have separate "methods" for each of those and then just aggregate them here.

Reply:
all right your call - I approved the thing anyway